worker/queue_wrapper.py

```diff
@@ -0,0 +1,54 @@
+"""
+Worker Queue
+"""
+
+import multiprocessing.managers
+import time
+import queue
+
+
+class QueueWrapper:
+    """
+    Wrapper for an underlying queue proxy which also stores max size.
+    """
+
+    __QUEUE_TIMEOUT = 0.1  # seconds
+    __QUEUE_DELAY = 0.1  # seconds
+
+    def __init__(self, mp_manager: multiprocessing.managers.SyncManager, max_size: int = 0) -> None:
+        self.queue = mp_manager.Queue(max_size)
+        self.max_size = max_size
+
+    def fill_queue_with_sentinel(self, timeout: float = 0.0) -> None:
+        """
+        Fills the queue with sentinel (None)
+        """
+        if timeout <= 0.0:
+            timeout = self.__QUEUE_TIMEOUT
+
+        try:
+            self.queue.put(None, timeout=timeout)
+            for _ in range(1, self.max_size):
+                self.queue.put(None, timeout=timeout)
+        except queue.Full:
+            return
+
+    def drain_queue(self, timeout: float = 0.0) -> None:
+        """
+        Drains the queue
+        """
+        if timeout <= 0.0:
+            timeout = self.__QUEUE_TIMEOUT
+
+        try:
+            self.queue.get(timeout=timeout)
+        except queue.Empty:
+            return
```

Comment:
this only removes 1 item from the queue, loop through the maxsize (like the fill function above) to completely drain the queue